code/datums/elements/haunted.dm

```diff
@@ -24,3 +24,27 @@
 	REMOVE_TRAIT(master, TRAIT_MOVE_FLYING, ELEMENT_TRAIT(type))
 	master.RemoveElement(/datum/element/movetype_handler)
 	return ..()
+
+/**
+ * Takes a given area and chance, applying the haunted_item component to objects in the area.
+ *
+ * Takes an epicenter, and within the range around it, runs a haunt_chance percent chance of
+ * applying the haunted_item component to nearby objects.
+ *
+ * * epicenter - The center of the outburst area.
+ * * range - The range of the outburst, centered around the epicenter.
+ * * haunt_chance - The percent chance that an object caught in the epicenter will be haunted.
+ */
+
+/proc/haunt_outburst(epicenter, range, haunt_chance, duration = 1 MINUTES)
+	var/effect_area = range(range, epicenter)
+	for(var/obj/item/object_to_possess in effect_area)
+		if(!prob(haunt_chance))
+			continue
+		object_to_possess.AddComponent(/datum/component/haunted_item, \
```

Comment:
This is in the file for /datum/**element**/haunted... either move this to the haunted component file or re-write this to work with the element.